lib/irb/ruby-lex.rb

```diff
@@ -65,6 +65,12 @@ def set_input(io, p = nil, context: nil, &block)
             false
           end
         else
+          # Accept any single-line input starting with a non-identifier alias (ex: @, $)
+          command = code.split(/\s/, 2).first
+          if context&.symbol_alias(command)
```

Comment:
Not directly related to this PR, but outside of tests the `context` should never be `nil`. [This](https://github.com/ruby/irb/blob/master/lib/irb.rb#L489) is its only caller of `set_input`, and it must not passing nil context, otherwise `@context.io` would already raise error.

Reply:
Thanks for https://github.com/ruby/irb/pull/427 :+1: Just pushed https://github.com/ruby/irb/pull/426/commits/ed0406fa479bd4da7847f98ed7ab891ffa1c4e70